### PR TITLE
Refactor polls frontend

### PR DESCRIPTION
**Pull request by Piwakk**
Use modern solutions for old problems

### CODE REVIEW

File: frontend/src/components/utils/table/sorting.ts [hunk 1]
```diff
@@ -30,14 +30,34 @@ export const cycleSorting = (current?: Sorting) =>
...
- * (in other words, multi-sort is _not_ supported).
+ * the value of the sorting parameter corresponding to the _first_ key containing either `Sorting.Ascending` or
+ * `Sorting.Descending` (in other words, multi-sort is _not_ supported).
+ *
+ * For instance:
+ * ```
+ * sortingToApiParameter(
+ *     {
+ *         myPiche: Sorting.Descending,
+ *         myPoche: Sorting.Ascending,
```
Comment (abocquet) [inline frontend/src/components/utils/table/sorting.ts:41]: is it a typo ?

File: frontend/src/components/utils/table/sorting.ts [hunk 2]
```diff
@@ -30,14 +30,34 @@ export const cycleSorting = (current?: Sorting) =>
...
+ *     },
+ *     {
+ *         myPiche: my_piche
```
Comment (abocquet) [inline frontend/src/components/utils/table/sorting.ts:44]: Who is `my_piche` here ?

File: frontend/src/components/utils/table/sorting.ts (hunk 1 above)
Comment (Piwakk) [reply frontend/src/components/utils/table/sorting.ts:41]: No, it's just for illustrating two different arguments

File: frontend/src/components/polls/table/PollsTable.tsx [hunk 3]
```diff
@@ -43,14 +43,20 @@ export const PollsTable = ({
...
+    const [stateFilter, setStateFilter] = useState<PollStateFilter>(
+        defaultStateFilter
+    );
+
+    if (!authService.isStaff && adminVersion) {
+        return <ForbiddenError />;
+    }
 
     return !authService.isStaff && adminVersion ? (
         <ForbiddenError />
```
[the remark this answers is not in the document]
Comment (Piwakk) [reply frontend/src/components/polls/table/PollsTable.tsx:62]: Indeed, merge problem I think

File: frontend/src/components/utils/table/sorting.ts (hunk 2 above)
Comment (Piwakk) [reply frontend/src/components/utils/table/sorting.ts:44]: See `mapping`: “an optional object which maps the key used by `sorting` to the keys used by the backend. Useful for
 * converting camelCase into snake_case.”

File: frontend/src/components/utils/table/sorting.ts (hunk 1 above)
Comment (abocquet) [reply frontend/src/components/utils/table/sorting.ts:41]: Then maybe use more different words to let no ambiguity

File: frontend/src/components/utils/table/sorting.ts (hunk 2 above)
Comment (abocquet) [reply frontend/src/components/utils/table/sorting.ts:44]: Ok, in that case maybe we could automatically convert from snake case to camel case?

**Pull request merged**